service/src/main/java/org/kiwiproject/champagne/model/DeployableSystemThreadLocal.java

```diff
@@ -0,0 +1,24 @@
+package org.kiwiproject.champagne.model;
+
+import java.util.Optional;
+
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class DeployableSystemThreadLocal {
+
+    private static final ThreadLocal<Long> THREAD_LOCAL = new ThreadLocal<>();
```

Comment:
Later when we get the new "scoped values" which will be previewed in Java 20, we should replace this. Obviously that's a ways off, but maybe it should be marked somewhere? But adding a TODO will be annoying since it will exist for a while. Would be nice to have a task list for each new JDK version, I.e. for 20 it could be on the list to replace all ThreadLocals with scoped values. Ideas on how or where to do that?

Reply:
Created a new ticket to track Java 20 things